### PR TITLE
Use latest master commit in gsctl build stage in container image

### PR DESCRIPTION
**Pull request by ubergesundheit**
This PR updates the Dockerfile to use the latest commit from master of `gsctl`. The required changes are in there.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,7 +2,6 @@
 FROM quay.io/giantswarm/golang:1.14.7 AS gsctl
 RUN git clone https://github.com/giantswarm/gsctl.git
 WORKDIR /go/gsctl
-RUN git checkout add-json-output
```
Comment (stone-z) [inline Dockerfile:5]: IMO we can skip the git clone and build altogether -- just pull the gsctl binary

File: Dockerfile [hunk 2]
```diff
@@ -2,7 +2,6 @@
...
 RUN CGO_ENABLED=0 go build
```
Comment (stone-z) [inline Dockerfile:5]: I suggest getting rid of all of this and pulling the binary directly in the alpine image

File: Dockerfile (hunk 1 above)
Comment (ubergesundheit) [reply Dockerfile:5]: with `go install`? The current changes in gsctl master are not in any release yet

Comment (tfussell) [reply Dockerfile:5]: I reckon this will work until gsctl is released with the json changes.

**Pull request merged**